internal/app/networkd/pkg/nic/netlink.go

```diff
@@ -85,6 +96,37 @@ func (n *NetworkInterface) configureBond(idx int, attrs *netlink.AttributeEncode
 	return nil
 }
 
+func (n *NetworkInterface) configureVlan(idx int, attrs *netlink.AttributeEncoder) error {
+	// Request the details of the interface
+	msg, err := n.rtConn.Link.Get(uint32(idx))
+	if err != nil {
+		return err
+	}
+
+	nlAttrBytes, err := attrs.Encode()
+	if err != nil {
+		return err
+	}
+	err = n.rtConn.Link.Set(&rtnetlink.LinkMessage{
+		Family: unix.AF_UNSPEC,
+		Type:   msg.Type,
+		Index:  msg.Index,
+		Change: 0,
+		Attributes: &rtnetlink.LinkAttributes{
+			Info: &rtnetlink.LinkInfo{
+				// https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/if_link.h#L612
+				Kind: "vlan",
+				Data: nlAttrBytes,
+			},
+		},
+	})
+	if err != nil {
```

Comment:
this could be simplified to `return n.rtConn.Link.Set(...)`

Reply:
Function removed as it is not used